packages/react-component-library/src/components/Timeline/Timeline.tsx

```diff
@@ -100,35 +107,32 @@ export const Timeline: React.FC<TimelineProps> = ({
     rangeInMonths: range || DEFAULTS.RANGE_IN_MONTHS,
   }
 
-  const bodyChildren = extractChildren(children, [TimelineRows.name])
-
-  const headChildren = extractChildren(children, [
-    TimelineDays.name,
-    TimelineWeeks.name,
-    TimelineMonths.name,
-    TimelineTodayMarker.name,
-  ])
-
   const rootChildren = extractChildren(
     children,
     [
-      TimelineRows.name,
       TimelineDays.name,
-      TimelineWeeks.name,
       TimelineMonths.name,
+      TimelineRows.name,
+      TimelineSide.name,
       TimelineTodayMarker.name,
+      TimelineWeeks.name,
     ],
     true
-  ).map((child, index) => {
-    if (isComponentOf(child, [TimelineSide.name])) {
-      return React.cloneElement(child, {
-        rowGroups: extractRowData(bodyChildren),
-        headChildren,
-        key: getKey('root-component', index),
-      })
-    }
-
-    return child
+  )
+
+  const hasTimelineSide = hasTimelineSideComponent(children)
+
+  const headChildren = extractChildren(children, [
+    TimelineDays.name,
+    TimelineWeeks.name,
+    TimelineMonths.name,
+    TimelineTodayMarker.name,
+  ])
+
+  const bodyChildren = extractChildren(children, [TimelineRows.name])
+
+  const innerClasses = classNames('timeline__inner', {
+    'timeline__inner--has-side': hasSide || hasTimelineSide,
```

Comment:
Could we combine the `hasSide || hasTimelineSide` into a single function or const so we don't have to repeat it?

Reply:
I actually thought about this at the time (and just looked at it again) but it appears that `hasSide || hasTimelineSide` is the neatest it can be, for two reasons:

1. `hasTimelineSide` is actually deprecated because use of the `TimelineSide` is deprecated which means this issue goes away once we properly get rid of `TimelineSide`
2. You could repurpose the `hasTimelineSideComponent` `function` but then once you do that, what do you call it? `hasSide` and `hasTimelineSide` are currently being used so then you end up with `getHasSide` which for the sake of replacing `hasSide || hasTimelineSide` it feels like the code is becoming less readable and then the issue goes away with point 1 above